marklogic-data-hub/src/main/java/com/marklogic/hub/step/StepRunnerFactory.java

```diff
@@ -33,7 +35,8 @@ public StepRunner getStepRunner(Flow flow, String stepNum) {
             .withBatchSize(step.getBatchSize())
             .withThreadCount(step.getThreadCount())
             .withSourceClient(hubConfig.newStagingClient(step.getSourceDatabase()))
-            .withDestinationDatabase(step.getDestinationDatabase());
+            .withDestinationDatabase(step.getDestinationDatabase())
+            .withOptions(step.getOptions());
```

Comment:
We don't have to set  step options, they are already available in the server. The only 'options' that has to be set is the one overridden from the command line which is ready set in FlowRunnerImpl

Reply:
@srinathgit I am confused about why we do not need step options? The configuration is deployed to ML server. We do not need them in DMSDK? 
How about switch problem?

Reply:
1. We only have to set the 'options' overriden from command line  in FlowRunner which is already done in https://github.com/marklogic/marklogic-data-hub/blob/feature/DHFPROD-1885-1889/marklogic-data-hub/src/main/java/com/marklogic/hub/flow/impl/FlowRunnerImpl.java#L212 . The other options are available in server and it will build a combined option using all of them.
2. We require the 'break' statement, since we are only running mapping step with flowrunner so far, it goes to the default case which returns a 'QueryStepRunner'

Reply:
@srinathgit thanks for quick response. Because of no break,  it went to QueryStepRunner. So I got https://project.marklogic.com/jira/browse/DHFPROD-2070 problem (my flow step is "INGEST",  not "mapping" type...)

Reply:
Currently, FlowRunner cannot run "INGEST" step , "WriteStepRunner" still requires work which will be done in this sprint.